### PR TITLE
Update Dockerfile to use Node alpine image instead of base alpine

### PR DESCRIPTION
**Pull request by gcoonrod**
The objective here is to reduce the amount of building that needs to happen when using Docker.

@evantahler how do I get rid of all these merge commits?

### CODE REVIEW

**Comment by evantahler** [comment]
That's a lot of merging :D

File: docker-compose.yml [hunk 1]
```diff
@@ -22,18 +22,33 @@ services:
     networks:
       - actionhero_frontend
       - actionhero_backend
+    volumes: 
+      - /actionhero/node_modules
+      - /actionhero/dist
```
Comment (evantahler) [inline docker-compose.yml:27]: What do volumes do?  I think that means they will persist on (each? one?) node in the docker cluster?

File: docker-compose.yml [hunk 2]
```diff
@@ -22,18 +22,33 @@ services:
...
   #
   # actionhero:
   #   build: ./
+  #   image: actionhero
   #   ports:
-  #     - 80:8080
+  #     - 8080
   #   environment:
-  #     TCP_PORTS: 8080
+  #     SERVICE_PORTS: 8080
```
Comment (evantahler) [inline docker-compose.yml:37]: Did HAPROXY change what it looks for?

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,11 @@
-FROM alpine:latest
-MAINTAINER admin@actionherojs.com
+FROM node:alpine
```
Comment (evantahler) [inline Dockerfile:1]: neat!

File: docker-compose.yml (hunk 1 above)
Comment (gcoonrod) [reply docker-compose.yml:27]: My understanding is that these volumes will ensure that any architecture differences that would impact module builds (i.e. node-gyp) between the host system and the container are eliminated by preventing these mounted directories from being shared with the host system. However, I'm no docker expert and these may not be strictly necessary.

File: docker-compose.yml (hunk 2 above)
Comment (gcoonrod) [reply docker-compose.yml:37]: I wasn't sure why the commented out build section differed from the uncommented section that pulls from docker hub.

**Comment by gcoonrod** [comment]
Closing in favor of #1583

**Pull request closed without merging**